common/lib/common-utils/src/instanceOf.ts

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Attempt to pin "load-time" implementations of call and toString to avoid
+ * behavior changes at "run-time" caused by unexpected prototype manipulation
+ */
+const pinnedObjectToString: (value: any) => string = Function.prototype.call.bind(
+    Object.prototype.toString, // eslint-disable-line @typescript-eslint/unbound-method
+);
+
+const toString: (value: any) => string = (value) => pinnedObjectToString(value);
```

Comment:
Maybe prototypeToString? It's a bit hard to read code like toString(value), as it looks like it should be equivalent to value.toString() but it's not.

Reply:
Oh ...today I learned that `toString` tends to exist in the global namespace already (because of `globalThis.constructor.prototype.toString`).  Definitely not a fan of "name shadowing" so I'll have to avoid this in the future -- thanks for the pointer.